<UserControl x:Class="NirDobovizki.MvvmControls.Wpf.Demo.Demos.ValidationDemo"
             xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
             xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
             xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006" 
             xmlns:d="http://schemas.microsoft.com/expression/blend/2008" 
             xmlns:local="clr-namespace:NirDobovizki.MvvmControls.Wpf.Demo.Demos"
             mc:Ignorable="d" 
             d:DesignHeight="300" d:DesignWidth="300"
             DataContext="{Binding RelativeSource={RelativeSource Self}}">
    <UserControl.Resources>
        <ResourceDictionary>
            <ResourceDictionary.MergedDictionaries>
                <ResourceDictionary Source="pack://application:,,,/MvvmControls.Wpf;component/Validation/ValidationTemplate.xaml"/>
            </ResourceDictionary.MergedDictionaries>
            <Style TargetType="TextBox">
                <Setter Property="Validation.ErrorTemplate" Value="{StaticResource ValidationTemplate}"/>
            </Style>
        </ResourceDictionary>
    </UserControl.Resources>
    <Grid>
        <StackPanel VerticalAlignment="Center" MaxWidth="150">
            <TextBlock TextWrapping="Wrap">Type something here then click the other edit control to throw an exception</TextBlock>
            <TextBox Text="{Binding AlwaysThrow, ValidatesOnExceptions=True}"/>
            <Rectangle Height="20"/>
            <TextBlock TextWrapping="Wrap">The text is the exception message, in case you are wandering</TextBlock>
            <Rectangle Height="20"/>
            <TextBlock TextWrapping="Wrap">This is here only so you have somewhere else to put focus:</TextBlock>
            <TextBox />

        </StackPanel>
    </Grid>
</UserControl>
